packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift

```diff
@@ -15,547 +16,509 @@ import XCTest
+     )
+
+     if let canEvaluateError = canEvaluateError {
+       error?.pointee = canEvaluateError
+       return false
+     }
+
+     return canEvaluatePolicyReturnValue
+   }
+
+   func evaluatePolicy(
+     _ policy: LAPolicy, localizedReason: String, reply: @escaping (Bool, (any Error)?) -> Void
+   ) {
+     XCTAssertEqual(
+       policy,
+       expectBiometrics
+         ? LAPolicy.deviceOwnerAuthenticationWithBiometrics
+         : LAPolicy.deviceOwnerAuthentication
+     )
+
+     DispatchQueue.global(qos: .background).async {
+       reply(self.evaluatePolicyResult.0, self.evaluatePolicyResult.1)
+     }
+   }
+ }
+
+final class StubAuthContext: NSObject, AuthContextProtocol {
+  var biometryType: LABiometryType = .none
   var expectBiometrics = false
-  /// The error to return from canEvaluatePolicy.
-  var canEvaluateError: NSError?
-  /// The value to return from evaluatePolicy:error:.
   var evaluateResponse = false
-  /// The error to return from evaluatePolicy:error:.
   var evaluateError: NSError?
 
-  // Overridden as read-write to allow stubbing.
-  var biometryType: LABiometryType = .none
-  var localizedFallbackTitle: String?
-
   func canEvaluatePolicy(_ policy: LAPolicy) throws {
-    XCTAssertEqual(
-      policy,
-      expectBiometrics
-        ? LAPolicy.deviceOwnerAuthenticationWithBiometrics
-        : LAPolicy.deviceOwnerAuthentication)
-    if let canEvaluateError = canEvaluateError {
-      throw canEvaluateError
+    if let error = evaluateError {
+      throw error
     }
   }
 
   func evaluatePolicy(
     _ policy: LAPolicy, localizedReason: String, reply: @escaping (Bool, Error?) -> Void
   ) {
-    XCTAssertEqual(
-      policy,
-      expectBiometrics
-        ? LAPolicy.deviceOwnerAuthenticationWithBiometrics
-        : LAPolicy.deviceOwnerAuthentication)
-    // evaluatePolicy:localizedReason:reply: calls back on an internal queue, which is not
-    // guaranteed to be on the main thread. Ensure that's handled correctly by calling back on
-    // a background thread.
     DispatchQueue.global(qos: .background).async {
       reply(self.evaluateResponse, self.evaluateError)
     }
   }
 }
 
-// MARK: -
+class MockAlertController: AlertController {
+  var showAlertCalled = false
+  var showAlertCompletionResult: Bool = false
+
+  func showAlert(
+    message: String,
+    dismissTitle dismissButtonTitle: String,
+    openSettingsTitle openSettingsButtonTitle: String?,
+    completion: @escaping (Bool) -> Void
+  ) {
+    showAlertCalled = true
+    completion(showAlertCompletionResult)
+  }
+}
 
 class FLALocalAuthPluginTests: XCTestCase {
+  var plugin: LocalAuthPlugin!
+  var mockAuthContext: MockAuthContext!
+  var mockAlertController: MockAlertController!
+
+  override func setUp() {
+    super.setUp()
+    mockAuthContext = MockAuthContext()
+    mockAlertController = MockAlertController()
+    plugin = LocalAuthPlugin(authContext: mockAuthContext, alertController: mockAlertController)
+  }
```

Comment:
Please revert these changes; we strongly prefer not having stateful tests, even at the expense of slightly more boilerplate in individual tests. https://abseil.io/tips/122 is a good resource on this topic (written for C++, but generall applicable).